src/components/MultiGesture.native.tsx

```diff
@@ -40,183 +44,149 @@ const gesture = (p1: Point, p2: Point, threshold: number) =>
-
-    this.panResponder = PanResponder.create({
-      // Prevent gesture when any text is selected.
-      // See https://github.com/cybersemics/em/issues/676.
-      // NOTE: thought it works simulating mobile on desktop, selectionchange is too late to prevent actual gesture on mobile, so we can't detect only when the text selection is being dragged
-      onMoveShouldSetPanResponder: () => !this.props.shouldCancelGesture?.() ?? true,
-      onMoveShouldSetPanResponderCapture: () => !this.props.shouldCancelGesture?.() ?? true,
-
-      // does not report moveX and moveY
-      // onPanResponderGrant: (e, gestureState) => {},
-
-      onPanResponderMove: (e: GestureResponderEvent, gestureState: GestureState) => {
-        if (this.abandon) {
-          return
-        }
-
-        if (this.props.shouldCancelGesture?.()) {
-          this.props.onCancel?.()
-          this.abandon = true
-          return
-        }
-
-        // use the first trigger of the move event to initialize this.currentStart
-        // onPanResponderStart does not work (why?)
-        if (!this.currentStart) {
-          this.scrolling = false
-          // ensure that disableScroll is false when starting in case it wasn't reset properly
-          // may be related to https://github.com/cybersemics/em/issues/1189
-          this.disableScroll = false
-          this.currentStart = {
-            x: gestureState.moveX,
-            y: gestureState.moveY,
-          }
-          this.scrollYStart = window.scrollY
-          if (this.props.onStart) {
-            this.props.onStart()
-          }
-          return
-        }
-
-        // abandon gestures when scrolling beyond vertical threshold
-        // because scrolling cannot be disabled after it has begin
-        // effectively only allows sequences to start with left or right
-        if (this.scrolling && Math.abs(this.scrollYStart! - window.scrollY) > this.props.scrollThreshold!) {
-          this.sequence = ''
-          this.props.onCancel?.()
-          this.abandon = true
-          return
-        }
-
-        const g = gesture(
-          this.currentStart,
-          {
-            x: gestureState.moveX,
-            y: gestureState.moveY,
+const MultiGesture = React.forwardRef<MultiGestureRef, MultiGestureProps>(
+  (
+    {
+      threshold = 3,
+
+      // the distance to allow scrolling before abandoning the gesture
+      scrollThreshold = 15,
+
+      // fired at the start of a gesture
+      // includes false starts
+      onStart = noop,
+
+      // fired when a new gesture is added to the sequence
+      onGesture = noop,
+
+      // fired when all gestures have completed
+      onEnd = noop,
+      ...props
+    },
+    ref,
+  ) => {
+    const [scrolling, setScroll] = useState(true)
+
+    let abandon = false
+    let currentStart: Point | null = null
+    let scrollYStart: number | null = null
+    let sequence: GesturePath = ''
+
+    useImperativeHandle(ref, () => ({ scrolling }))
```

Comment:
I think using ref is not necessary. The `MultiGesture` can prevent the `ScrollView` from scrolling when the gesture is active. Since the `MultiGesture` wraps the `Content`, the nested scroll would be prevented as well. Whenever the gesture is cancelled or abandoned, `ScrollView` can be allowed to scroll again. Please let me know if I am missing something here.

Reply:
Because we have a nested scroll view we also need to prevent the children scrolling otherwise we'd still have some scroll interfering with the gestures.

Reply:
Just had a second look, and if we really need to prevent the scroll on gestures we also will need to pass `isGestureActive` to its children being `Content` which is a scrollable view, and its parent `ScrollView`. This way `Content` would then receive a prop to manage its scrolling.

Reply:
Got it!